src/nvim/ops.c

```diff
@@ -1585,6 +1586,10 @@ int op_delete(oparg_T *oap)
 
   msgmore(curbuf->b_ml.ml_line_count - old_lcount);
 
+  ++textlock;
```

Comment:
In this case I would suggest replaces `pre{increment,decrement}` which `post{increment,decrement}` even if this is just porting.

see: https://neovim.io/develop/style-guide.xml#Postincrement_and_Postdecrement